File: lib/models/perfil_financeiro.dart
/// As respostas do onboarding.
///
/// [origem] diz se os números vieram de chute ou de extrato — muda a
/// confiança que o app pode ter neles.
enum OrigemPerfil { feeling, ofx }

class PerfilFinanceiro {
  const PerfilFinanceiro({
    this.nome,
    this.rendaMensal,
    this.diaRenda,
    this.gastoDiarioEstimado,
    this.contasFixasEstimadas,
    this.origem = OrigemPerfil.feeling,
  });

  /// Como a pessoa quer ser chamada. O design abre o Início com
  /// "bom dia, Gabriel" — sem isso o cabeçalho fica sem dono.
  /// Nulo é aceito: aí o app só diz "bom dia".
  final String? nome;

  final double? rendaMensal;
  final int? diaRenda;
  final double? gastoDiarioEstimado;
  final double? contasFixasEstimadas;
  final OrigemPerfil origem;

  bool get veioDeExtrato => origem == OrigemPerfil.ofx;

  PerfilFinanceiro copyWith({
    String? nome,
    double? rendaMensal,
    int? diaRenda,
    double? gastoDiarioEstimado,
    double? contasFixasEstimadas,
    OrigemPerfil? origem,
  }) =>
      PerfilFinanceiro(
        nome: nome ?? this.nome,
        rendaMensal: rendaMensal ?? this.rendaMensal,
        diaRenda: diaRenda ?? this.diaRenda,
        gastoDiarioEstimado: gastoDiarioEstimado ?? this.gastoDiarioEstimado,
        contasFixasEstimadas: contasFixasEstimadas ?? this.contasFixasEstimadas,
        origem: origem ?? this.origem,
      );
}

File: app/Models/PerfilFinanceiro.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

/**
 * Respostas do onboarding (spec §3.3). Uma linha por usuário.
 */
class PerfilFinanceiro extends Model
{
    use HasFactory;

    protected $table = 'perfil_financeiro';

    public const ORIGEM_FEELING = 'feeling';
    public const ORIGEM_OFX = 'ofx';

    protected $fillable = [
        'user_id',
        'renda_mensal',
        'dia_renda',
        'gasto_diario_estimado',
        'contas_fixas_estimadas',
        'origem',
    ];

    protected function casts(): array
    {
        return [
            'renda_mensal' => 'decimal:2',
            'gasto_diario_estimado' => 'decimal:2',
            'contas_fixas_estimadas' => 'decimal:2',
        ];
    }

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    /** true quando os números vieram de extrato, não de chute. */
    public function veioDeExtrato(): bool
    {
        return $this->origem === self::ORIGEM_OFX;
    }
}
